### ISSUE TITLE
Build issue on Windows 10, Visual Studio 2022

### ISSUE THREAD

**Issue by FrankRay78** [annacrombie/plot#28]
Build output when performing `meson build`

![image](https://user-images.githubusercontent.com/52075808/221210141-97b90486-1625-4562-8d41-4e8497e5d038.png)

`meson.build` file with the offending line:

![image](https://user-images.githubusercontent.com/52075808/221210366-4f357300-3ce4-426c-8da4-3b6052e9087d.png)

Any ideas what library 'm' is, and how I might be able to fix this error?

**Comment by FrankRay78** [comment]
My mistake.  Because this implementation of asciichart is written in C, and also because it uses meson for the build (which is cross-platform), I mistakenly believed the codebase would naturally be cross-platform as well.  It turns out that it has been written for Linux and not Windows.

An example is the use of `sigaction` in animate.c, which is Linux only.

![image](https://user-images.githubusercontent.com/52075808/221286805-728afbe6-f296-4719-b8ce-0943855edf2a.png)

Compiling it on Windows results in the following build error:

![image](https://user-images.githubusercontent.com/52075808/221286908-e1e36fe1-5841-4993-8fa2-a57ab98758f0.png)

The code would need to change with a conditional statement to include platform specific code, as per:

![image](https://user-images.githubusercontent.com/52075808/221287078-6f784edf-8777-4bba-854a-153e0a516769.png)

ref:  https://medium.com/codex/running-open-source-software-on-windows-50dd1231cae9

The other problem I see is also the missing getopts C library/functions which also are not included on windows.

**Comment by annacrombie** [comment]
Correct, plot has not been tested on windows, and I don't have much knowledge of windows.

**Comment by FrankRay78** [comment]
I’m going to see if I can get it working on windows, I don’t think the code changes will be too extensive. If I do manage it, I’ll submit a PR for your consideration. I’d really like to be able to use plot in my cross-platform app, if possible. Nice work on the port btw.

**Comment by FrankRay78** [comment]
Hello @annacrombie, I think I'm pretty close to having plot compile on both Linux and Windows now. I've persisted with this as largely a learning exercise, and it seems the Microsoft C compiler has a few known quirks when it comes to the meson build system.  Can I please ask if you remember what Linux distros / compilers you used when developing plot?  If you remember, I'd be interested in testing my changes on these to make sure I haven't introduced a regression somewhere.

**Comment by annacrombie** [comment]
I primarily develop on alpine linux with gcc.  This is a pretty standard setup aside from the fact that alpine linux uses musl libc rather than glibc, which means that some non-standard extensions that some programs expect are _not_ present.

One other thing since it seems you are intending on submitting a PR -- I prefer doing platform specific code without using #ifdefs when possible.  One strategy I use a lot is this:

```meson
src = files(...)

if platform == 'windows'
  src += 'src/platform/windows/example.c'
else
  src += 'src/platform/posix/example.c'
endif
```

```c
// include/platform/example.h

// Define some common api functions

void do_this(int);
```

```c
// src/platform/windows/example.c

void
do_this(int)
{
  // windows-specific code
}
```

```c
// src/platform/posix/example.c

void
do_this(int)
{
  // posix-specific code
}
```

**Comment by sskras** [comment]
@FrankRay78, did you test your changes on Alpine?

@annacrombie, do you use gcc or clang on it? The CI uses gcc on Ubuntu, it seems:
https://github.com/annacrombie/plot/blob/784ec84b328a2ec30565a7aa3ac6e959be580cf7/.github/workflows/ci.yml#L16

**Comment by FrankRay78** [comment]
> [@FrankRay78](https://github.com/FrankRay78), did you test your changes on Alpine?

No, I didn't. I was trying to build it on Windows 10.
